### DISCUSSION TITLE
Adapt rules on introspection relations ?

### DISCUSSION THREAD

**Discussion by olup** [prisma/prisma#11901] [Q&A] [answered]
In our DB, we have a table `order` with a field `integration_origin` - text - that has a foreign key on the `value` field of a table `integration_origin`. This is a design required by Hasura to generate enums in graphql.

The generated relation in the schema is : `integration_origin_integration_originToorders`. 

With the client, we cannot set `integration_origin`. as a string, we have to connect in the mentioned relation. But the name is problematic for the DX of our team, discoverability, and onboarding.

Is there a way to change to relations name in the `db pull` process? Some post-introspection hook at least to change the schema programmatically? Or should I just overwrite it with some sed command?

**Comment by janpio** [comment, marked as answer]
Right now Introspection can not be configured out of the box. You can of course run a script yourself, and future Re-Introspections of the same database should keep the manual changes in the Prisma schema (Open a bug if they don't!) 

There are a few related issues under https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22topic%3A+introspection-configuration%22, but if your specific request is not in there, just open a new feature request please.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
